### PR TITLE
Fix for vanishing titles

### LINKED ISSUE #119: Vanishing titles
Iohannes' report: 
>there is (and it was already here) a bug that makes some of these titles "vanish". E.g. Ptolemy the Liberator is scheduled to gain k_egypt, but instead he is reduced to a count. I have no idea why this happens. It is not the order of the file, as Chandragupta, with the same format, keeps his e_india.

### PR DESCRIPTION
**Pull request by IhateTrains**
Fixes #119

### CODE REVIEW

File: ImperatorToCK3/Source/CK3/Titles/LandedTitles.cpp [hunk 1]
```diff
@@ -41,26 +43,32 @@ std::optional<std::string> CK3::LandedTitles::getCountyForProvince(const unsigne
 	for (const auto& [titleName, title] : foundTitles)
 	{
 		if (titleName.starts_with("c_") && !title->getCountyProvinces().empty())
-			if (title->getCountyProvinces().contains(provinceID)) return titleName;
+			if (title->getCountyProvinces().contains(provinceID))
```
Comment (Zemurin) [inline ImperatorToCK3/Source/CK3/Titles/LandedTitles.cpp:46]: No need to nest ifs, you can put the conditional right above after additional &&. If the second one fails, third one won't execute and won't crash. In this particular case no need to poke if empty before poking with contains, compiler should be smart enough to return false if container is empty.

File: ImperatorToCK3/Source/CK3/Titles/Title.cpp [hunk 2]
```diff
@@ -160,7 +166,24 @@ void CK3::Title::initializeFromTag(std::shared_ptr<Imperator::Country> theCountr
 	
 	// --------------- Adjective Locs
 	trySetAdjectiveLoc(localizationMapper);
+}
+
+void CK3::Title::updateFromTitle(const std::shared_ptr<Title>& otherTitle)
+{
+	if (titleName != otherTitle->titleName)
+		throw std::runtime_error(titleName + " can not be updated from  " + otherTitle->titleName + ": different title names!");
```
Comment (Zemurin) [inline ImperatorToCK3/Source/CK3/Titles/Title.cpp:174]: (often best to throw a warning and return, stuff happens, inputs are iffy)

**Review by Zemurin — APPROVED** [review]

**Review by Idhrendur — APPROVED** [review]

File: ImperatorToCK3/Source/CK3/Titles/LandedTitles.cpp (hunk 1 above)
Comment (IhateTrains) [reply ImperatorToCK3/Source/CK3/Titles/LandedTitles.cpp:46]: Yeah contains() returns false if the container is empty, so the empty() is redundant.

File: ImperatorToCK3/Source/CK3/Titles/Title.cpp (hunk 2 above)
Comment (IhateTrains) [reply ImperatorToCK3/Source/CK3/Titles/Title.cpp]: Replaced with error level log.

**Pull request merged**